src/components/vanilla/charts/TableChart/index.tsx

```diff
@@ -1,54 +1,56 @@
 import { DataResponse, DimensionOrMeasure, OrderBy, OrderDirection } from '@embeddable.com/core';
 import { useEmbeddableState } from '@embeddable.com/react';
-import React, { useCallback, useEffect, useLayoutEffect, useMemo, useRef, useState } from 'react';
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
 
 import formatValue from '../../../util/format';
 import Container from '../../Container';
-import { ChevronLeft, ChevronRight, SortDown, SortUp } from '../../icons';
+import { SortDown, SortUp } from '../../icons';
+import debounce from 'lodash.debounce';
+import Pagination from './components/Pagination';
+import TableHead from './components/TableHead';
+import { REGULAR_FONT_SIZE } from '../../../constants';
+import { SortDirection } from '../../../../enums/SortDirection';
 
 export type Props = {
   limit?: number;
   results: DataResponse;
   defaultSort?: { property: DimensionOrMeasure; direction: string }[];
   columns: DimensionOrMeasure[];
   title: string;
+  fontSize?: string;
+  minColumnWidth?: number;
 };
 
 type Meta = { page: number; maxRowsFit: number; sort: OrderBy[] };
 
 type Record = { [p: string]: string };
 
 export default (props: Props) => {
-  const ref = useRef<HTMLDivElement | null>(null);
   const { columns, results } = props;
   const [maxRowsFit, setMaxRowFit] = useState(0);
 
-  useLayoutEffect(() => {
-    let val = 0;
-
-    const interval = setInterval(() => {
-      const elem = ref.current?.parentElement?.parentElement;
-      const heightWithoutHead = (elem?.clientHeight || 72) - 72;
-      const newMaxRowsFit = Math.floor(heightWithoutHead / 44);
-
-      if (
-        (maxRowsFit === newMaxRowsFit && newMaxRowsFit === val) ||
-        props.results?.data?.length === 0
-      ) {
-        return;
-      }
-      setMaxRowFit((val = newMaxRowsFit));
-    }, 100);
-
-    return () => clearInterval(interval);
-  }, [maxRowsFit, props.results]);
-
   const [meta, setMeta] = useEmbeddableState({
     page: 0,
     maxRowsFit: 0,
     sort: props.defaultSort
   }) as [Meta, (f: (m: Meta) => Meta) => void];
 
+  const calculateMaxRowFix = useCallback(({ height }: { height: number }) => {
+    let val = 0;
+
+    const heightWithoutHead = (height || 72) - 72;
+    const newMaxRowsFit = Math.floor(heightWithoutHead / 44);
+
+    if (
+      (maxRowsFit === newMaxRowsFit && newMaxRowsFit === val) ||
+      props.results?.data?.length === 0
+    ) {
+      return;
+    }
+    setMaxRowFit((val = newMaxRowsFit));
+
+  }, [maxRowsFit, props.results]);
+
```

Comment:
@kuka-radovan I sometimes end up with a situation where the pagination at the bottom appears slightly off the page. Are we able to fix this?
<img width="360" alt="Screenshot 2024-08-05 at 11 41 31" src="https://github.com/user-attachments/assets/59e02517-1a2f-4928-840a-ec814bcb7c56">

Reply:
@kuka-radovan additionally, I've found an issue where, if you make the table column very narrow, it wraps the text, which makes the row height higher, and means fewer rows than expected fit. Let's discuss this as it's not currently clear to me what the solution should be (e.g. should we not wrap the text vs re-calculating the height of the table dynamically (based on each row) once it's already loaded and re-rendering).

https://github.com/user-attachments/assets/b24171dd-f836-40f5-ade9-42e7ee520d9d

Reply:
I am not sure how to deal with this. Calculation of max row fit assumes a fixed line height of 44 pixels. In case of wrapped text, row height will be bigger. I could return classes to ellipsis overflowed text.

Reply:
> @kuka-radovan I sometimes end up with a situation where the pagination at the bottom appears slightly off the page. Are we able to fix this? <img alt="Screenshot 2024-08-05 at 11 41 31" width="360" src="https://private-user-images.githubusercontent.com/150246125/355055067-59e02517-1a2f-4928-840a-ec814bcb7c56.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjI5NTI3OTEsIm5iZiI6MTcyMjk1MjQ5MSwicGF0aCI6Ii8xNTAyNDYxMjUvMzU1MDU1MDY3LTU5ZTAyNTE3LTFhMmYtNDkyOC04NDBhLWVjODE0YmNiN2M1Ni5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwODA2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDgwNlQxMzU0NTFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05YjgwN2UxOTQ1M2QxNjBhZjllODJjMWEzMzE5MThkZWZmYzJjODk5YmQwOTAwOTk0MzYwNjJlMGVlYjRjYjNlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.Gb5n5DDpbL1cmpB_bRMzjBFNg3rWgGdXhXjp_W3sgTA">

I didn't change the logic of calculating max row fit. As I look at it now, the table component assumes that the parent component (which is a card component) has title and description defined. It takes the height of this card and subtracts 72 pixels (hardcoded) from it (height with description should be 72 pixels). However, if your table does not have title or description, this problem will occur.

Reply:
> I am not sure how to deal with this. Calculation of max row fit assumes a fixed line height of 44 pixels. In case of wrapped text, row height will be bigger. I could return classes to ellipsis overflowed text.

Done

Reply:
> > @kuka-radovan I sometimes end up with a situation where the pagination at the bottom appears slightly off the page. Are we able to fix this? <img alt="Screenshot 2024-08-05 at 11 41 31" width="360" src="https://private-user-images.githubusercontent.com/150246125/355055067-59e02517-1a2f-4928-840a-ec814bcb7c56.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjI5NTI3OTEsIm5iZiI6MTcyMjk1MjQ5MSwicGF0aCI6Ii8xNTAyNDYxMjUvMzU1MDU1MDY3LTU5ZTAyNTE3LTFhMmYtNDkyOC04NDBhLWVjODE0YmNiN2M1Ni5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwODA2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDgwNlQxMzU0NTFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05YjgwN2UxOTQ1M2QxNjBhZjllODJjMWEzMzE5MThkZWZmYzJjODk5YmQwOTAwOTk0MzYwNjJlMGVlYjRjYjNlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.Gb5n5DDpbL1cmpB_bRMzjBFNg3rWgGdXhXjp_W3sgTA">
> 
> I didn't change the logic of calculating max row fit. As I look at it now, the table component assumes that the parent component (which is a card component) has title and description defined. It takes the height of this card and subtracts 72 pixels (hardcoded) from it (height with description should be 72 pixels). However, if your table does not have title or description, this problem will occur.

Fixed. Caused by added top margin.